aws_grant_user_access/src/process_event.py

```diff
@@ -70,3 +71,29 @@ def publish_sns_message(message: SNSMessage) -> None:
         SNSMessagePublisher(config.get_sns_client()).publish_sns_message(
             sns_topic_arn=sns_topic_arn, message=json.dumps(message.to_dict())
         )
+
+
+def validate_request(event: Dict[str, Any]) -> str:
+    logger = Config.configure_logging()
+    iam_client = config.get_iam_client()
+    user_names = event["usernames"]
+    role_arn = event["role_arn"]
+    # check time range is valid
+    if not 1 <= event["approval_in_hours"] <= 8760:
+        retmsg = f"Invalid time period specified: {event['approval_in_hours']} hours. Valid input is 1-8760 hours"
+        logger.info(retmsg)
+        return retmsg
+    # check role is an engineer role
```

Comment:
Are these comments needed? Seems like they're covered by the error messages but if not maybe these could be extracted as individual validation functions?